docs/aws/audit/kubernetesmonitoring/rules/is_ecr_vuln_scan_enabled.mdx

```diff
@@ -24,6 +24,243 @@ AWSWAF, HITRUST, SOC2, NISTCSF, PCIDSS
+
+   ```sh
+   aws iam attach-role-policy --role-name <your-role-name> --policy-arn arn:aws:iam::<your-account-id>:policy/EnforceECRImageScanning
+   ```
+
+   Or for a user:
+
+   ```sh
+   aws iam attach-user-policy --user-name <your-user-name> --policy-arn arn:aws:iam::<your-account-id>:policy/EnforceECRImageScanning
+   ```
+
+4. **Automate Scanning with a Lambda Function (Optional):**
+   Optionally, you can create a Lambda function that automatically triggers image scans for existing images in the repository. This can be set up to run periodically using CloudWatch Events.
+
+   ```sh
+   aws lambda create-function --function-name TriggerECRImageScan --runtime python3.8 --role arn:aws:iam::<your-account-id>:role/<your-lambda-role> --handler lambda_function.lambda_handler --zip-file fileb://function.zip
+   ```
+
+   Ensure the Lambda function code includes logic to trigger image scans for existing images.
+
+By following these steps, you can ensure that image vulnerability scanning is enabled and enforced for your Amazon ECR repositories in Kubernetes using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent image vulnerability scanning misconfigurations for Amazon Elastic Container Registry (ECR) in Kubernetes using Python scripts, you can follow these steps:
+
+### 1. **Install Required Libraries**
+First, ensure you have the necessary libraries installed. You will need `boto3` to interact with AWS services.
+
+```bash
+pip install boto3
+```
+
+### 2. **Set Up AWS Credentials**
+Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+```bash
+aws configure
+```
+
+### 3. **Enable Image Scanning on ECR Repositories**
+Use a Python script to enable image scanning on your ECR repositories. This script will iterate through all your ECR repositories and enable image scanning.
+
+```python
+import boto3
+
+def enable_image_scanning():
+    # Create a session using your AWS credentials
+    session = boto3.Session(
+        aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+        aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+        region_name='YOUR_AWS_REGION'
+    )
+
+    # Create an ECR client
+    ecr_client = session.client('ecr')
+
+    # List all ECR repositories
+    response = ecr_client.describe_repositories()
+    repositories = response['repositories']
+
+    # Enable image scanning for each repository
+    for repo in repositories:
+        repo_name = repo['repositoryName']
+        ecr_client.put_image_scanning_configuration(
+            repositoryName=repo_name,
+            imageScanningConfiguration={
+                'scanOnPush': True
+            }
+        )
+        print(f"Enabled image scanning for repository: {repo_name}")
+
+if __name__ == "__main__":
+    enable_image_scanning()
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>